llama_index/llms/monsterapi.py

```diff
@@ -121,25 +124,65 @@ def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
 
     @llm_completion_callback()
     def complete(
-        self, prompt: str, formatted: bool = False, **kwargs: Any
+        self, prompt: str, formatted: bool = False, timeout: int = 100, **kwargs: Any
     ) -> CompletionResponse:
         if not formatted:
             prompt = self.completion_to_prompt(prompt)
 
+        stream = kwargs.pop("stream", False)
+
+        if stream is True:
+            raise NotImplementedError(
+                "complete method cannot be used with stream=True, please use stream_complete method"
+            )
+
         # Validate input args against input Pydantic model
         input_dict = self._get_input_dict(prompt, **kwargs)
 
-        # Send request and receive process_id
-        response = self._client.get_response(model=self.model, data=input_dict)
-        process_id = response["process_id"]
+        result = self._client.generate(
+            model=self.model, data=input_dict, timeout=timeout
+        )
+
+        if isinstance(result, Exception):
+            raise result
+
+        if isinstance(result, dict) and "error" in result:
+            raise RuntimeError(result["error"])
 
-        # Wait for response and return result
-        result = self._client.wait_and_get_result(process_id)
+        if isinstance(result, dict) and "text" in result:
+            if isinstance(result["text"], list):
+                return CompletionResponse(text=result["text"][0])
+            elif isinstance(result["text"], str):
+                return CompletionResponse(text=result["text"])
 
-        return CompletionResponse(text=result["text"])
+        if isinstance(result, list):
+            return CompletionResponse(text=result[0]["text"])
+
+        raise RuntimeError("Unexpected Return please contact monsterapi support!")
 
     @llm_completion_callback()
-    def stream_complete(
-        self, prompt: str, formatted: bool = False, **kwargs: Any
-    ) -> CompletionResponseGen:
-        raise NotImplementedError
+    def stream_complete(self, prompt: str, **kwargs: Any) -> CompletionResponseGen:
+        if "deploy" not in self.model:
+            raise NotImplementedError(
+                "stream_complete method can only be used with deploy models for now. Support for other models will be added soon."
+            )
+
+        # Validate input args against input Pydantic model
+        input_dict = self._get_input_dict(prompt, **kwargs)
+        input_dict["stream"] = True
+
+        # Starting the stream
+        result_stream = self._client.generate(model=self.model, data=input_dict)
+
+        if isinstance(result_stream, Exception):
+            raise result_stream
+
+        if isinstance(result_stream, dict) and "error" in result_stream:
+            raise RuntimeError(result_stream["error"])
+
+        # Iterating over the generator
+        try:
+            for result in result_stream:
+                yield CompletionResponse(text=result[0])
```

Comment:
Does monster api return the newest token here, or the full text each time?

The `CompletionResponse` object has two main fields -- a `text` and `delta`.
`text` == the most current complete version of the streamed text
`dela` == the newest addition to the streamed text

Reply:
@Vikasqblocks

Reply:
Full token each time!